BIC_codes/functions/dFC_funcs.py

```diff
@@ -0,0 +1,3472 @@
+            ContinuousHMM
+            Windowless
+            Clustering_pear_corr
+    '''
+
+    if name_lst_key is None:
+        fig_width = 25*(len(data)/10)
+    else:
+        fig_width = 35*(len(data)/10) + 4
+    fig_height = 10
+
+    fig, axs = plt.subplots(1, len(data), figsize=(fig_width, fig_height), \
+        facecolor='w', edgecolor='k')
+
+    if not type(axs) is np.ndarray:
+        axs = np.array([axs])
+
+    fig.suptitle(title) #, fontsize=20, size=20
+
+    axs = axs.ravel()
+
+    for i, key in enumerate(data):
+
+        name_lst = None
+        if not name_lst_key is None:
+            if type(name_lst_key) is str:
+                name_lst = data[key][name_lst_key]
+            if type(name_lst_key) is list:
+                name_lst = name_lst_key
+
+        if mat_key is None:
+            C = data[key]
+        else:
+            C = data[key][mat_key]
+
+        # C = np.abs(C) # ?????? should we do this?
+
+        if normalize:
+            C = dFC_mat_normalize(C[None,:,:], global_normalization=False, threshold=0.0)[0]
+
+        if not disp_diag:
+            C = np.multiply(C, 1-np.eye(len(C)))
+            C = C + np.mean(C.flatten()) * np.eye(len(C))
+
+        if np.any(C<0): # ?????? should we do this?
+            V_MIN = -1
+            V_MAX = 1
+        else: # ?????? should we do this?
+            V_MIN = 0
+            V_MAX = lim_val
+
+        if not fix_lim:
+            V_MAX = np.max(C)
+            V_MIN = np.min(C)
+
+        if name_lst is None:
+            axs[i].set_axis_off()
+
+        im = axs[i].imshow(C, interpolation='nearest', aspect='equal', cmap=cmap,    # 'viridis' or 'jet'
+            vmin=V_MIN, vmax=V_MAX)
+        
+        if not name_lst is None:
+            axs[i].set_xticks(np.arange(len(name_lst)))
+            axs[i].set_yticks(np.arange(len(name_lst)))
+            axs[i].set_xticklabels(name_lst, rotation=90, fontsize=9)
+            axs[i].set_yticklabels(name_lst, fontsize=9)
+        axs[i].set_title(key)
+
+    fig.subplots_adjust(
+        bottom=0.1, \
+        top=1.5, \
+        left=0.1, \
+        right=0.9,
+        # wspace=0.02, \
+        # hspace=0.02\
+    )
+
+    if not name_lst is None:
+        fig.subplots_adjust(
+            wspace=0.85 
+        )
+        
+    if name_lst is None:
+        cb_ax = fig.add_axes([0.91, 0.75, 0.007, 0.1])
+    else:
+        cb_ax = fig.add_axes([0.91, 0.75, 0.02, 0.1])
+    cbar = fig.colorbar(im, cax=cb_ax, shrink=0.8) # shrink=0.8??
+
+    # # set the colorbar ticks and tick labels
+    # cbar.set_ticks(np.arange(0, 1.1, 0.5))
+    # cbar.set_ticklabels(['0', '0.5', '1'])
+
+    if save_image:
+        folder = output_root[:output_root.rfind('/')]
+        if not os.path.exists(folder):
+            os.makedirs(folder)
+        plt.savefig(output_root+'.png', \
+            dpi=fig_dpi, bbox_inches=fig_bbox_inches, pad_inches=fig_pad \
+        ) 
+        plt.close()
+    else:
+        plt.show()
+
+'''
```

Comment:
is this to comment out code?
trust version control and delete the code you don't need instead of commenting it out